src/components/general/subjectIdToAttrDrv.js

```diff
@@ -0,0 +1,71 @@
+/**
+ *  @directive subjectIdToAttrDrv
+ *  This directive is an evolution of 'subjectIdToClassDrv'
+ *  @context-attr a comma separated string of attribute names
+ *  @znk-prefix a comma separated string of prefixes to the attribute values
+ *  @znk-suffix a comma separated string of suffixes to the attribute values
+ *
+ *  In case only one prefix/suffix is provided, it will be used in all attributes
+ *  In case no @context-attr is provided, it will set the class attribute by default
+ *  No need to pass dashes ('-') to prefix or suffix, they are already appended
+ */
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra.general').directive('subjectIdToAttrDrv', [
+        'SubjectEnum',
+        function (SubjectEnum) {
+            return {
+                scope: {
+                    contextAttr: '@',
+                    znkPrefix: '@',
```

Comment:
i added znk- because i thought attr names such as "prefix" and "suffix" are too generic. what do you say?